### ISSUE TITLE
Creating / Renaming Card Title JS Errors

### ISSUE THREAD

**Issue by joshfleming** [nextcloud/deck#754]
### Steps to reproduce
1. Click title of existing card
1. Type new title
1. Press Enter

### Expected behaviour
Title changes on the screen, with no JS errors. 

### Actual behaviour
The title doesn't change on the screen until the browser is reloaded.

## Server configuration detail

**Operating system:** Linux 4.15.0-39-generic #42-Ubuntu SMP Tue Oct 23 15:48:01 UTC 2018 x86_64

**Webserver:** Apache/2.4.25 (Debian) (apache2handler)

**Database:** pgsql PostgreSQL 11.1 (Debian 11.1-1.pgdg90+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516, 64-bit

**PHP version:** 

7.2.12
Modules loaded: Core, date, libxml, openssl, pcre, sqlite3, zlib, ctype, curl, dom, fileinfo, filter, ftp, hash, iconv, json, mbstring, SPL, PDO, session, posix, Reflection, standard, SimpleXML, pdo_sqlite, Phar, tokenizer, xml, xmlreader, xmlwriter, mysqlnd, apache2handler, apcu, exif, gd, intl, ldap, memcached, pcntl, pdo_mysql, pdo_pgsql, redis, sodium, zip, Zend OPcache

**Nextcloud version:** 14.0.4 - 14.0.4.2

**Updated from an older Nextcloud/ownCloud or fresh install:** 

**Where did you install Nextcloud from:** 
This is a Docker container from https://hub.docker.com/_/nextcloud/ 
It's behind an Nginx reverse proxy container.

<details><summary>Signing status</summary>

Array
(
)

</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.0.1
 - activity: 2.7.0
 - calendar: 1.6.4
 - circles: 0.15.2
 - cloud_federation_api: 0.0.1
 - comments: 1.4.0
 - contacts: 2.1.8
 - dav: 1.6.0
 - deck: 0.5.0
 - encryption: 2.2.0
 - event_update_notification: 0.3.1
 - federatedfilesharing: 1.4.0
 - federation: 1.4.0
 - files: 1.9.0
 - files_antivirus: 1.4.2
 - files_mindmap: 0.0.10
 - files_pdfviewer: 1.3.2
 - files_sharing: 1.6.2
 - files_texteditor: 2.6.0
 - files_trashbin: 1.4.1
 - files_versions: 1.7.1
 - files_videoplayer: 1.3.0
 - gallery: 18.1.0
 - issuetemplate: 0.4.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.2.0
 - nextcloud_announcements: 1.3.0
 - notifications: 2.2.1
 - oauth2: 1.2.1
 - onlyoffice: 2.1.0
 - password_policy: 1.4.0
 - polls: 0.8.3
 - provisioning_api: 1.4.0
 - serverinfo: 1.4.0
 - socialsharing_email: 1.0.4
 - spreed: 4.0.1
 - survey_client: 1.2.0
 - systemtags: 1.4.0
 - theming: 1.5.0
 - twofactor_backupcodes: 1.3.1
 - twofactor_totp: 1.5.0
 - updatenotification: 1.4.1
 - workflowengine: 1.4.0
Disabled:
 - admin_audit
 - dashboard
 - files_external
 - firstrunwizard
 - sharebymail
 - support
 - user_external
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "htaccess.RewriteBase": "\/",
    "memcache.local": "\\OC\\Memcache\\APCu",
    "apps_paths": [
        {
            "path": "\/var\/www\/html\/apps",
            "url": "\/apps",
            "writable": false
        },
        {
            "path": "\/var\/www\/html\/custom_apps",
            "url": "\/custom_apps",
            "writable": true
        }
    ],
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "readyroom.revoltmedia.com"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "pgsql",
    "version": "14.0.4.2",
    "overwrite.cli.url": "https:\/\/readyroom.revoltmedia.com",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "updater.release.channel": "stable",
    "updater.secret": "***REMOVED SENSITIVE VALUE***",
    "maintenance": false
}
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** 1

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

## Client configuration

**Browser:** Mozilla/5.0 (X11; Linux x86_64; rv:62.0) Gecko/20100101 Firefox/62.0

**Operating system:** Debian 9

## Logs

<details><summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

***Creating a new card POST***
Title doesn't change until reloading the page. The following Errors show in JS console: 

vendor.js?v=aac0bf61-12:7
/apps/deck/cards:1 Failed to load resource: the server responded with a status of 500 ()
vendor.js?v=aac0bf61-12:7 Possibly unhandled rejection: Fetchingcards failed
(anonymous) @ vendor.js?v=aac0bf61-12:7

  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
-- | -- | -- | --
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | $digest | @ | vendor.js?v=aac0bf61-12:7
  | $apply | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | dispatch | @ | core.js?v=aac0bf61-12:3
  | r.handle | @ | core.js?v=aac0bf61-12:3

***Renaming a Card PUT***
vendor.js?v=aac0bf61-12:7 PUT https://domain.tld/apps/deck/cards/7 500
(anonymous) @ vendor.js?v=aac0bf61-12:7
(anonymous) @ vendor.js?v=aac0bf61-12:7
(anonymous) @ vendor.js?v=aac0bf61-12:7
(anonymous) @ vendor.js?v=aac0bf61-12:7
(anonymous) @ vendor.js?v=aac0bf61-12:7
$digest @ vendor.js?v=aac0bf61-12:7
$apply @ vendor.js?v=aac0bf61-12:7
(anonymous) @ vendor.js?v=aac0bf61-12:7
dispatch @ core.js?v=aac0bf61-12:3
r.handle @ core.js?v=aac0bf61-12:3

vendor.js?v=aac0bf61-12:7 
Possibly unhandled rejection: Updating cards failed

  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
-- | -- | -- | --
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | $digest | @ | vendor.js?v=aac0bf61-12:7
  | $apply | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | dispatch | @ | core.js?v=aac0bf61-12:3
  | r.handle | @ | core.js?v=aac0bf61-12:3

Network Request for /apps/deck/cards

Request URL: https://readyroom.revoltmedia.com/apps/deck/cards
Request Method: POST
Status Code: 500 
Remote Address: 68.183.248.67:443
Referrer Policy: no-referrer
cache-control: no-store, no-cache, must-revalidate
content-length: 5485
content-security-policy: default-src 'self'; script-src 'self' 'unsafe-eval' 'nonce-UkxaMUxIS2lJaVRhbGkyWGU4SjgvT2VoZ0xidElFOHpXZmp6WVFvYVAxOD06TU1NdFl3cU5WVVhwNUZqVFZMZ3lpSkR6OVlLVVNDaFNQNzZ4VW5rcVJ5az0='; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *; object-src 'none'; base-uri 'self';
content-type: text/html; charset=UTF-8
date: Sun, 02 Dec 2018 02:27:58 GMT
expires: Thu, 19 Nov 1981 08:52:00 GMT
pragma: no-cache
server: nginx
status: 500
strict-transport-security: max-age=31536000
x-content-type-options: nosniff
x-download-options: noopen
x-frame-options: SAMEORIGIN
x-permitted-cross-domain-policies: none
x-powered-by: PHP/7.2.12
x-robots-tag: none
x-xss-protection: 1; mode=block
:authority: readyroom.revoltmedia.com
:method: POST
:path: /apps/deck/cards
:scheme: https
accept: application/json, text/plain, */*
accept-encoding: gzip, deflate, br
accept-language: en-US,en;q=0.9
content-length: 46
content-type: application/json;charset=UTF-8
cookie: oc_sessionPassphrase=TMkYCnShZJba%2BvHwooLENQxJDCzyuHDkmNLGFtfabnfj6lZEaui%2FlARwxZGgelTIwrapHBChCDbFzi4R8KY%2FEjIuKhejOUJtMdQEFiqwiQEGaThdaCCq0RsMlIBTizYk; __Host-nc_sameSiteCookielax=true; __Host-nc_sameSiteCookiestrict=true; oc67z4et1xsz=9ae6d500100775ead0fb63d6ca71d35f; nc_username=operador; nc_token=dPoLD%2FdtE1FRVY4AApbeLHj0SxQlfmOE; nc_session_id=9ae6d500100775ead0fb63d6ca71d35f
origin: https://readyroom.revoltmedia.com
requesttoken: YIZyyZKhft/HRluyx6NEHwVZRPHX3YgEmBWxHGLX4dU=:FPMqhuqOCb70NC726NkKa3ILMcWute9l/lPzLxHnmaM=
user-agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36
{title: "asdfas3", stackId: 4, type: "plain"}
stackId: 4
title: "asdfas3"
type: "plain"

```
</details>

<details><summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

<details><summary>Browser log</summary>

Insert your browser log here, this could for example include:

***Creating a new card POST***
Title doesn't change until reloading the page. The following Errors show in JS console: 

vendor.js?v=aac0bf61-12:7
/apps/deck/cards:1 Failed to load resource: the server responded with a status of 500 ()
vendor.js?v=aac0bf61-12:7 Possibly unhandled rejection: Fetchingcards failed
(anonymous) @ vendor.js?v=aac0bf61-12:7

  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
-- | -- | -- | --
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | $digest | @ | vendor.js?v=aac0bf61-12:7
  | $apply | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | dispatch | @ | core.js?v=aac0bf61-12:3
  | r.handle | @ | core.js?v=aac0bf61-12:3

***Renaming a Card PUT***
vendor.js?v=aac0bf61-12:7 PUT https://domain.tld/apps/deck/cards/7 500
(anonymous) @ vendor.js?v=aac0bf61-12:7
(anonymous) @ vendor.js?v=aac0bf61-12:7
(anonymous) @ vendor.js?v=aac0bf61-12:7
(anonymous) @ vendor.js?v=aac0bf61-12:7
(anonymous) @ vendor.js?v=aac0bf61-12:7
$digest @ vendor.js?v=aac0bf61-12:7
$apply @ vendor.js?v=aac0bf61-12:7
(anonymous) @ vendor.js?v=aac0bf61-12:7
dispatch @ core.js?v=aac0bf61-12:3
r.handle @ core.js?v=aac0bf61-12:3

vendor.js?v=aac0bf61-12:7 
Possibly unhandled rejection: Updating cards failed

  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
-- | -- | -- | --
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | $digest | @ | vendor.js?v=aac0bf61-12:7
  | $apply | @ | vendor.js?v=aac0bf61-12:7
  | (anonymous) | @ | vendor.js?v=aac0bf61-12:7
  | dispatch | @ | core.js?v=aac0bf61-12:3
  | r.handle | @ | core.js?v=aac0bf61-12:3

Network Request for /apps/deck/cards

Request URL: https://readyroom.revoltmedia.com/apps/deck/cards
Request Method: POST
Status Code: 500 
Remote Address: 68.183.248.67:443
Referrer Policy: no-referrer
cache-control: no-store, no-cache, must-revalidate
content-length: 5485
content-security-policy: default-src 'self'; script-src 'self' 'unsafe-eval' 'nonce-UkxaMUxIS2lJaVRhbGkyWGU4SjgvT2VoZ0xidElFOHpXZmp6WVFvYVAxOD06TU1NdFl3cU5WVVhwNUZqVFZMZ3lpSkR6OVlLVVNDaFNQNzZ4VW5rcVJ5az0='; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *; object-src 'none'; base-uri 'self';
content-type: text/html; charset=UTF-8
date: Sun, 02 Dec 2018 02:27:58 GMT
expires: Thu, 19 Nov 1981 08:52:00 GMT
pragma: no-cache
server: nginx
status: 500
strict-transport-security: max-age=31536000
x-content-type-options: nosniff
x-download-options: noopen
x-frame-options: SAMEORIGIN
x-permitted-cross-domain-policies: none
x-powered-by: PHP/7.2.12
x-robots-tag: none
x-xss-protection: 1; mode=block
:authority: readyroom.revoltmedia.com
:method: POST
:path: /apps/deck/cards
:scheme: https
accept: application/json, text/plain, */*
accept-encoding: gzip, deflate, br
accept-language: en-US,en;q=0.9
content-length: 46
content-type: application/json;charset=UTF-8
cookie: oc_sessionPassphrase=TMkYCnShZJba%2BvHwooLENQxJDCzyuHDkmNLGFtfabnfj6lZEaui%2FlARwxZGgelTIwrapHBChCDbFzi4R8KY%2FEjIuKhejOUJtMdQEFiqwiQEGaThdaCCq0RsMlIBTizYk; __Host-nc_sameSiteCookielax=true; __Host-nc_sameSiteCookiestrict=true; oc67z4et1xsz=9ae6d500100775ead0fb63d6ca71d35f; nc_username=operador; nc_token=dPoLD%2FdtE1FRVY4AApbeLHj0SxQlfmOE; nc_session_id=9ae6d500100775ead0fb63d6ca71d35f
origin: https://readyroom.revoltmedia.com
requesttoken: YIZyyZKhft/HRluyx6NEHwVZRPHX3YgEmBWxHGLX4dU=:FPMqhuqOCb70NC726NkKa3ILMcWute9l/lPzLxHnmaM=
user-agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36
{title: "asdfas3", stackId: 4, type: "plain"}
stackId: 4
title: "asdfas3"
type: "plain"

</details>

**Comment by putt1ck** [comment]
Testing on a similar setups (Ubuntu/Postgres/LDAP), but installed onto a base OS, one on 15RC1 and the other on 14.03, both accessed via an Apache reverse proxy, I cannot reproduce this issue. Tried

Click on title of existing card and amend then press enter;
click on title, delete, type new title and then press enter.

In both cases the title updates in both the "main" view and the details pane, no delay.

Maybe an Nginx config?

**Comment by juliushaertl** [comment]
Can you check your nextcloud log during the failing requests? They should contain more information on what's going wrong.

**Comment by joshfleming** [comment]
```Doctrine\DBAL\Exception\TableNotFoundException: An exception occurred while executing 'SELECT s.board_id as id, c.stack_id as stack_id FROM oc_deck_stacks as s inner join oc_deck_cards as c ON c.stack_id = s.id WHERE c.id = ?' with params [16]: SQLSTATE[42P01]: Undefined table: 7 ERROR: relation "oc_deck_stacks" does not exist LINE 1: ...ECT s.board_id as id, c.stack_id as stack_id FROM oc_deck_st... ^```

This error shows in the Nextcloud log when attempting to edit a title. Could this be related to a on my database user with not enough permissions to create the table? Or is this something else?

**Comment by joshfleming** [comment]
I tried switching to the root Postgres root user so it doesn't seem like the problem is related to the DB user permissions.

**Comment by bpcurse** [comment]
It is probably https://github.com/nextcloud/deck/issues/705 again, as a dbtableprefix other than "oc_" is set in config. @joshfleming Try to apply the fix proposed in #705.

**Comment by joshfleming** [comment]
Yes, Modifying that line fixed it. However that issue is closed and states that bug only exists in NC 13 and I'm running NC14. Is this fix queued up for an upcoming release?

**Comment by bpcurse** [comment]
@joshfleming AFAIK the bugfix will be shipped in 0.5.1
